lib/redis/cluster/slot.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'set'
+
+class Redis
+  class Cluster
+    # Keep slot and node key map for Redis Cluster Client
+    class Slot
+      def initialize(available_slots, node_flags = {}, with_replica = false)
+        @map = build_slot_node_key_map(available_slots, node_flags, with_replica)
+      end
+
+      def exists?(slot)
+        @map.key?(slot)
+      end
+
+      def not_exists?(slot)
```

Comment:
Not sure this is very useful, I'd rather see `unless slot.exist?(...)` than `if slot.not_exist(...)`